flink-streaming-java/src/test/java/org/apache/flink/streaming/util/typeutils/FieldAccessorTest.java

```diff
@@ -368,4 +369,23 @@ public void testIllegalBasicType2() {
 
 		FieldAccessor<Long, Long> f = FieldAccessorFactory.getAccessor(tpeInfo, "foo", null);
 	}
+
+	/**
+	 * Validates that no ClassCastException happens
+	 * should not fail e.g. like in FLINK-8255.
+	 */
+	@Test
+	public void testRowTypeInfo() {
```

Comment:
This test just validates that a `FieldAccessor` is created. At runtime it would fail with a `ClassCastException`.